### PR TITLE
Fix reference-type parameter handling in search

### PR DESCRIPTION
**Pull request by sachintha-lk**
## Purpose
Fix incorrect handling of reference-type search parameters during search queries.                                                                                                          

Provide a fix for: https://github.com/wso2-enterprise/open-healthcare/issues/2091

Note that this fix can be improved in a more robust approach in the future after implementing https://github.com/wso2-enterprise/open-healthcare/issues/2096

## Goals
- Identify reference-type search parameters by querying SEARCH_PARAM_RES_EXPRESSIONS instead  of relying on a `/` heuristic in the parameter value.
- Prevent non-reference parameters (e.g. code, category) from being incorrectly routed to the REFERENCES table when their value contains a / . 
- Prevent skipping of reference param filtering in search if `/` is missed in parameter value.
- Return a not-supported message for parameter values that are not supported yet ( e.g. : `subject=<id>`, `subject=<absolute-path>`)

## Approach
Add a lookup against SEARCH_PARAM_RES_EXPRESSIONS table to determine whether a search parameter is of type reference. 

## Automation tests
 - Integration tests
     - Added reference search test cases to `tests/integration-tests.sh` covering correct reference queries, malformed values, non-reference params with Type/id values, and plain-id references.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved classification and parsing of search parameters, with stricter validation and clearer rejections for malformed reference formats.
  * Enhanced reference resolution with type-constraint checks and safer ID filtering/escaping.
  * More specific error responses for invalid reference searches (returning 400) and clearer messages for search failures.

* **Tests**
  * Added integration tests covering reference-parameter searches, validation edge cases, type-constraint behavior, and cleanup.

* **Chores**
  * Increased server startup readiness polling timeout.

### CODE REVIEW

File: miscellaneous/fhir-server/modules/utils/commons.bal [hunk 1]
```diff
@@ -19,7 +19,8 @@ const string LAST_UPDATED_COLUMN = "LAST_UPDATED";
 
 // Escape single quotes in SQL string values to prevent SQL injection
 public isolated function escapeSql(string value) returns string {
-    return regex:replaceAll(value, "'", "''");
+    string escaped = regex:replaceAll(value, "'", "''");
+    return regex:replaceAll(escaped, "%", "\\%");
 }
```
[the remark this answers is not in the document]
Comment (sachintha-lk) [reply miscellaneous/fhir-server/modules/utils/commons.bal:23]: outdated comment

File: miscellaneous/fhir-server/modules/mappers/read_mapper.bal [hunk 2]
```diff
@@ -1140,25 +1167,92 @@ public class ReadMapper {
         return revIncludedEntries;
     }
 
-    // Check if a search parameter is a custom extension parameter
-    private isolated function isCustomSearchParam(jdbc:Client jdbcClient, string resourceType, string paramName) returns boolean|error {
+
+    // Parse "resolve() is X" from a expression in SEARCH_PARAM_RES_EXPRESSIONS table and return X, or () if not present.
+    // Example: "Condition.subject.where(resolve() is Patient)" → "Patient"
```
Comment (sameeragunarathne) [inline miscellaneous/fhir-server/modules/mappers/read_mapper.bal:1172]: Can we represent this in the table column such as as reference types and have a list? For every search doing regex evaluation considered to be costly. Need to find a way to optimize this

**Review by sameeragunarathne — APPROVED** [review]
